velox/expression/SignatureBinder.cpp

```diff
@@ -163,6 +163,12 @@ bool SignatureBinderBase::checkOrSetIntegerParameter(
     return false;
   }
 
+  const auto& constraint = variables().at(parameterName).constraint();
+  if (isPositiveInteger(constraint) && atoi(constraint.c_str()) != value) {
+    // Return false if the actual value does not match the constraint.
+    return false;
+  }
```

Comment:
Hi @rui-mo, let's add a unit test for this change since it not only affects fuzzer. By the way, since checkOrSetIntegerParameter() is used in both ReverseSignatureBinder and the regular SignatureBinder, let's add unit tests for both. The rest of this PR looks good to me.

Reply:
I updated this PR with corresponding tests. Could you please have another look? Thanks!